### ISSUE TITLE
Bug: 'Automatically create missing options' creates doublicates

### ISSUE THREAD

**Issue by kaurov** [BlackbitDigitalCommerce/pimcore-data-director#29]
Select-boxes (Pimcore\Model\DataObject\ClassDefinition\Data\Select::class) have checkbox 'Automatically create missing options' in <>Settings on AttributeMapping tab.

That is great feature. But it creates doublicate values for some values (not all, some are ok).

Buggy output sample:
```
array (
  0 =>
  Pimcore\Model\DataObject\ClassDefinition\Data\Select::__set_state(array(
     'fieldtype' => 'select',
     'options' =>
    array (
      0 =>
      array (
        'key' => 'Einzelpreis',
        'value' => '0',
      ),

      // auto generated values below
      1 =>
      array (
        'key' => '0',
        'value' => '0',
      ),
      2 =>
      array (
        'key' => '0',
        'value' => '0',
      ),
      3 =>
      array (
        'key' => '0',
        'value' => '0',
      ),
      4 =>
      array (
        'key' => '0',
        'value' => '0',
      ),
   ...  
```

One more sample:
```
 array (
                      0 => 
                      Pimcore\Model\DataObject\ClassDefinition\Data\Select::__set_state(array(
                         'fieldtype' => 'select',
                         'options' => 
                        array (
                          0 => 
                          array (
                            'key' => 'Bundesrepublik Deutschland',
                            'value' => 'DE',
                          ),
                          1 => 
                          array (
                            'key' => 'Indonesia',
                            'value' => 'ID',
                          ),
                          2 => 
                          array (
                            'key' => 'Spanien',
                            'value' => 'ES',
                          ),

                         // following values are generated automatically
                          3 => 
                          array (
                            'key' => 'DE',
                            'value' => 'DE',
                          ),
                          4 => 
                          array (
                            'key' => 'DE',
                            'value' => 'DE',
                          ),
                        ),
```

PS: New Feature Request: add the ability to specify from which field to take the key and from wchi to take the value for new options creation.

**Comment by BlackbitNeueMedien** [comment]
Thanks for your report but I cannot reproduce this. I created a select field with one option: label: "Einzelpreis", value: "0"

Then I created an import with these settings for the select field:
<img width="1231" alt="Bildschirmfoto 2022-05-18 um 07 59 31" src="https://user-images.githubusercontent.com/8749138/168967814-ad46c55b-53be-4e5e-a6f1-a6ed97be428e.png">

And it did assign "Einzelpreis" as expected.

Can you please explain what you did different?

**Comment by kaurov** [comment]
If you will create Objectbrick and add it to some Object class then definition_Object.php would be changed but Object brick definition would not be changed.
It means that next Object import attempt will not find necessary option (it really does not exists in Objectbrick) and will doublicate it in definition_Object.php

Expected behaviour: the new option should be added in select-box of ObjectBrick, not Class using it.

**Comment by BlackbitNeueMedien** [comment]
Is fixed in 3.0.7
Problem only occured when the select field was inside object brick, field collection or classification store (or in other words: not directly in the data object class).
